### PR TITLE
[artifact-manager] (#331) Fix CRITICAL and HIGH level vulnerabilities

### PR DESCRIPTION
**Pull request by bcpmihail**
### Description

Trivy identified critical and high level vulnerabilities in the spring framework and snakeyaml dependencies for artifact-manager.
These were resolved by using the latest versions of com.fasterxml.jackson (of which snakeyaml is a transitive dependency) and org.springframework; this required upgrading org.apache.httpcomponents to 5.3, affecting: 
- HttpClient initialization
- usage of HttpStatus
- BasicAuthorizationInterceptor (replaced by BasicAuthenticationInterceptor)
- handling of URISyntaxException and ConfigurationException

After the upgrade, no more critical or high level vulnerabilities were detected by trivy.

### Checklist

- [ ] I have added relevant error handling and logging messages to help troubleshooting
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation, relevant usage information (if applicable)
- [x] I have updated the PR title with affected component, related issue number and a short summary of the changes introduced
- [x] I have added labels for implementation kind (kind/*) and version type (version/*)
- [ ] I have tested against live environment, if applicable
- [ ] I have synced any structure and/or content vRA-NG improvements with vra-ng and ts-vra-ng archetypes (if applicable)
- [x] I have my changes rebased and squashed to the minimal number of relevant commits. **Notice: don't squash all commits**
- [x] I have added a descriptive commit message with a short title, including a `Fixed #XXX -` or `Closed #XXX -` prefix to auto-close the issue

### Testing

Builds and deploys test project successfully to environment.

### Related issues and PRs

https://github.com/vmware/build-tools-for-vmware-aria/issues/331

### CODE REVIEW

**Comment by Michaelpalacce** [comment]
Wow, so this is great effort and a great job. Honestly, The only problem is that there are so many irrelevant changes in the form of convention and formatting, that the important ones get easily lost. I am not sure I can do a good review of this :/

Tests are passing, and they are unchanged, outside some imports, so that is good and pretty much good enough. Not sure why but again no matter what IDE I use Java dev on BTVA is.... something else... Trying to resolve that and will continue my efforts

**Comment by bcpmihail** [comment]
> Wow, so this is great effort and a great job. Honestly, The only problem is that there are so many irrelevant changes in the form of convention and formatting, that the important ones get easily lost. I am not sure I can do a good review of this :/
> 
> Tests are passing, and they are unchanged, outside some imports, so that is good and pretty much good enough. Not sure why but again no matter what IDE I use Java dev on BTVA is.... something else... Trying to resolve that and will continue my efforts

Shall I overwrite the git history to separate the style changes (formatting) from the rest for better readability? You'll need to review commit by commit afterwards.

**Comment by bcpmihail** [comment]
Currently addressing the codeql issues (mainly adding Javadoc comments)

File: maven/plugins/vrealize/src/main/java/com/vmware/pscoe/maven/plugins/PushMojo.java [hunk 1]
```diff
@@ -54,30 +54,45 @@ public class PushMojo extends AbstractIacMojo {
...
-		return PackageFactory.getInstance(PackageType.fromExtension(artifact.getType()), artifact.getFile(), new MavenArtifactPackageInfoProvider(artifact).getPackageName());
+		return PackageFactory.getInstance(PackageType.fromExtension(artifact.getType()), artifact.getFile(),
+				new MavenArtifactPackageInfoProvider(artifact).getPackageName());
 	}
 
-	private void importArtifacts(Collection<Artifact> allArtifacts) throws MojoExecutionException {
-		Map<String, List<Artifact>> artifactsByType = allArtifacts.stream().collect(Collectors.groupingBy(Artifact::getType));
-		
+	private void importArtifacts(Collection<Artifact> allArtifacts)
+			throws MojoExecutionException, ConfigurationException {
```
Comment (Michaelpalacce) [inline maven/plugins/vrealize/src/main/java/com/vmware/pscoe/maven/plugins/PushMojo.java:62]: God, this ConfigurationException seems like a complete mess... Can't we do `ConfigurationException extends RuntimException` and be done with it?

File: maven/plugins/vrealize/src/main/java/com/vmware/pscoe/maven/plugins/PushMojo.java [hunk 2]
```diff
@@ -54,30 +54,45 @@ public class PushMojo extends AbstractIacMojo {
...
 				continue;
 			}
 			try {
-				List<Package> packages = artifactsByType.get(type.getKey()).stream().map(PushMojo::packageFromArtifact).collect(Collectors.toList());
+				List<Package> packages = artifactsByType.get(type.getKey()).stream().map(PushMojo::packageFromArtifact)
+						.collect(Collectors.toList());
 				PackageStore<?> store = getConfigurationForType(PackageType.fromExtension(type.getKey()))
-						.flatMap(configuration -> Optional.of(PackageStoreFactory.getInstance(configuration))).orElseThrow(() -> new ConfigurationException(
-								"Unable to find PackageStore based on configuration. Make sure there is configuration for type: " + pkgType.name()));
+						.flatMap(configuration -> PackageStoreFactory.getInstanceWrapped(configuration))
```
Comment (Michaelpalacce) [inline maven/plugins/vrealize/src/main/java/com/vmware/pscoe/maven/plugins/PushMojo.java:75]: I get it, the whole idea behind getInstanceWrapped. TBF, I prefer it (using RuntimeException), :D Can't we do it everywhere?

Comment (Michaelpalacce) [reply maven/plugins/vrealize/src/main/java/com/vmware/pscoe/maven/plugins/PushMojo.java:75]: I am just desperate to clean up this "Error Propagation" or whatever you want to call it

File: common/artifact-manager/src/main/java/com/vmware/pscoe/iac/artifact/PackageStoreFactory.java [hunk 3]
```diff
@@ -57,25 +58,45 @@ private PackageStoreFactory() {
...
+	 * Regarding the strategies, the following rules apply:
+	 * 
+	 * - If forceImportLatestVersions is set, the StrategyForceLatestVersions is
+	 * used and the importOldVersions flag is ignored.
+	 *
+	 * @param configuration The configuration object.
+	 * @param <T>           The configuration type.
+	 * @return The PackageStore instance as Optional.
+	 */
+	public static <T extends Configuration> Optional<PackageStore<?>> getInstanceWrapped(T configuration) {
```
Comment (Michaelpalacce) [inline common/artifact-manager/src/main/java/com/vmware/pscoe/iac/artifact/PackageStoreFactory.java:73]: Cool change!

**Review by Michaelpalacce — CHANGES_REQUESTED** [review]
Haven't looked fully at the primitives and the clients, but let's fix the Config part, and we'll worry about that later. If you agree, of course.

**Comment by Michaelpalacce** [comment]
Also forgot to mention. Let's ignore the linter. Let it complain. Last thing we want is more to this PR.

File: common/artifact-manager/src/main/java/com/vmware/pscoe/iac/artifact/rest/RestClientVrops.java [hunk 4]
```diff
@@ -543,7 +563,8 @@
 				exportHeader.set(INTERNAL_API_HEADER_NAME, Boolean.TRUE.toString());
 			}
 			HttpEntity<String> exportEntity = new HttpEntity<>(exportHeader);
-			ResponseEntity<byte[]> exportResponse = restTemplate.exchange(uriBuilder.toUriString(), HttpMethod.GET, exportEntity, byte[].class);
+			ResponseEntity<byte[]> exportResponse = restTemplate.exchange(uriBuilder.toUriString(), HttpMethod.GET,
```
[the remark this answers is not in the document]
Comment (bcpmihail) [reply common/artifact-manager/src/main/java/com/vmware/pscoe/iac/artifact/rest/RestClientVrops.java:566]: The URI comes from configuration (ConfigurationVrops)

**Comment by bcpmihail** [comment]
> Also forgot to mention. Let's ignore the linter. Let it complain. Last thing we want is more to this PR.

Changed ConfigurationException to inherit from RuntimeException.
Removed unnecessary changes in error handling.
Edited git history to format all affected files in a single commit - easy to see actual changes in subsequent commits.
Temporarily disabled linter checks.
Will need help to temporarily comment out last 2 rows in codeql-java.yml.
Will open additional PR after merging this one to revert the temporary changes.

**Comment by bcpmihail** [comment]
Addressed linter issues* and reverted changes to sun_checks.xml.
* except Javadocs/DesignForExtension - they are unrelated to the changes and too many. As discussed with @Michaelpalacce, the PR can be force-merged without fixing them.

**Comment by Michaelpalacce** [comment]
Can you sync with master @bcpmihail.

**Comment by bcpmihail** [comment]
> Can you sync with master @bcpmihail.

@Michaelpalacce  Done

**Pull request merged**